src/ApiCore/GapicClientTrait.php

```diff
@@ -162,6 +172,26 @@ private function buildClientOptions(array $options)
             'rest' => [],
         ];
 
+        if (isset($defaultOptions['gcpApiConfigPath'])
```

Comment:
Is it intentional that this acts on `$defaultOptions` instead of on `$options`? I would actually expect this logic to live somewhere like line 312. That way, if a user passes in a value of `gcpApiConfigPath`, or if an API updates the value by overriding `modifyClientOptions()`, then that value would be used. With this implementation, only the value defined in `getClientDefaults()` in the generated client will ever be used, with no way to update it.

Reply:
I use $defaultOptions because the actual 'gcpApiConfigPath' is initialized in getClientDefaults. See https://github.com/googleapis/gapic-generator/pull/2231 . It should be defined in the client library, we don't actually want the user to pass in this value.

As for the modifyClientOptions(), I think you are right. The client should be able to use this method to override the path value.

Thanks!

Reply:
After reading the code, I think a better place to put this block of logic is line 175, right before the `buildClientOptions()` returns. Because to me `setClientOptions()` is more like a read only operations on options array, but we are adding a new key to that array. So I would rather keep this logic inside `buildClientOptions()` but after `modifyClientOptions()`.

Reply:
SGTM

Reply:
done.